### PR TITLE
Add Fall Damage Replay Recording

### PR DESCRIPTION
**Pull request by sagarg22**
Adds a new recording for later use to test for fall damage.

### CODE REVIEW

File: src/test/java/org/terasology/replayTests/FallDamageTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import org.terasology.engine.GameThread;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.logic.health.HealthComponent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.recording.RecordAndReplayStatus;
+import org.terasology.registry.CoreRegistry;
+import org.terasology.world.WorldProvider;
+
+import static org.junit.Assert.assertEquals;
```
Comment (iaronaraujo) [inline src/test/java/org/terasology/replayTests/FallDamageTest.java]: Unused Import

File: src/test/java/org/terasology/replayTests/FallDamageTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import org.terasology.ReplayTestingEnvironment;
+import org.terasology.TestUtils;
+import org.terasology.engine.GameThread;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.logic.health.HealthComponent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.recording.RecordAndReplayStatus;
+import org.terasology.registry.CoreRegistry;
+import org.terasology.world.WorldProvider;
```
Comment (iaronaraujo) [inline src/test/java/org/terasology/replayTests/FallDamageTest.java]: Unused Import

File: src/test/java/org/terasology/replayTests/FallDamageTest.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+import org.junit.After;
+import org.junit.Test;
+import org.terasology.ReplayTestingEnvironment;
+import org.terasology.TestUtils;
+import org.terasology.engine.GameThread;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.logic.health.HealthComponent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.math.geom.Vector3i;
```
Comment (iaronaraujo) [inline src/test/java/org/terasology/replayTests/FallDamageTest.java]: Unused Import

**Review by iaronaraujo — CHANGES_REQUESTED** [review]
Hello, Phoenix, thank you very much for your PR! The code seems pretty solid, there were only a few unused imports that I would like you to remove! Besides that, I am having problems executing the replay tests after the module.txt change. Did you have any problems executing the tests after it as well? If not, did you do something else besides run gradlew idea after you modified it?

File: src/test/java/org/terasology/replayTests/FallDamageTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        replayThread.join();
+    }
+
+    @Test
+    public void testFallDamage() {
+
+        replayThread.start();
+
+        LocalPlayer localPlayer = CoreRegistry.get(LocalPlayer.class);
+        TestUtils.waitUntil(() -> localPlayer.isValid());
```
Comment (llvieira) [inline src/test/java/org/terasology/replayTests/FallDamageTest.java]: You should wait for the environment to be initialized first. So the right order is:

```java
TestUtils.waitUntil(() -> (environment.isInitialised() && environment.getRecordAndReplayStatus() == RecordAndReplayStatus.REPLAYING));
TestUtils.waitUntil(() -> localPlayer.isValid());
```

Without this change I'm getting the following error message:

```java
23:27:57.860 [main] ERROR org.terasology.TestUtils - Error has occurred in the waitUntil method: 
java.lang.NullPointerException: null
	at org.terasology.replayTests.FallDamageTest.lambda$testFallDamage$0(FallDamageTest.java:60)
	at org.terasology.TestUtils.waitUntil(TestUtils.java:32)
	at org.terasology.replayTests.FallDamageTest.testFallDamage(FallDamageTest.java:60)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
```

**Review by llvieira — CHANGES_REQUESTED** [review]

File: src/test/java/org/terasology/replayTests/FallDamageTest.java (hunk 4 above)
Comment (sagarg22) [reply src/test/java/org/terasology/replayTests/FallDamageTest.java]: The change has been made!

**Comment by llvieira** [comment]
@FieryPhoenix909 There are still one weird error that is not allowing the game to run. Here is the complete log: https://hastebin.com/loqulubasu.sql. We are investigating it. Do you have any idea why it is happening? On your machine everything is working OK, right?

File: src/test/java/org/terasology/replayTests/FallDamageTest.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        GameThread.reset();
+        replayThread.join();
+    }
+
+    @Test
+    public void testFallDamage() {
+
+        replayThread.start();
+
+        LocalPlayer localPlayer = CoreRegistry.get(LocalPlayer.class);
```
Comment (iaronaraujo) [inline src/test/java/org/terasology/replayTests/FallDamageTest.java]: This line should be between the "waitUntil" (line 60 and 61)

**Review by iaronaraujo — CHANGES_REQUESTED** [review]

**Review by llvieira — APPROVED** [review]

**Pull request merged**